extensions/rtmp/src/main/java/io/github/thibaultbee/streampack/ext/rtmp/streamers/CameraDualLiveStreamer.kt

```diff
@@ -0,0 +1,406 @@
+import io.github.thibaultbee.streampack.internal.sources.AudioSource
+import io.github.thibaultbee.streampack.internal.sources.camera.CameraSource
+import io.github.thibaultbee.streampack.listeners.OnConnectionListener
+import io.github.thibaultbee.streampack.listeners.OnErrorListener
+import io.github.thibaultbee.streampack.logger.Logger
+import io.github.thibaultbee.streampack.streamers.settings.BaseCameraStreamerSettings
+import io.github.thibaultbee.streampack.streamers.settings.BaseStreamerAudioSettings
+import kotlinx.coroutines.runBlocking
+import java.io.File
+import java.nio.ByteBuffer
+import java.util.UUID
+
+
+class CameraDualLiveStreamer(
+    private val context: Context,
+    initialOnErrorListener: OnErrorListener? = null,
+    initialOnConnectionListener: OnConnectionListener? = null
+) : EventHandler() {
+
+    private enum class EncoderIndex(val index: Int) {
+        RTMP(0),
+        FILE(1)
+    }
+
+    private val rtmpProducer =
+        RtmpProducer().apply { onConnectionListener = initialOnConnectionListener }
+    private val fileWriter = FileWriter()
+    private var onErrorListener: OnErrorListener? = initialOnErrorListener
+
+    private val audioSource = AudioSource()
+    private val cameraSource = CameraSource(context)
+    private val rtmpMuxer: IMuxer = FlvMuxer(writeToFile = false)
+    private val fileMuxer: IMuxer = MP4Muxer()
+
+    private var rtmpAudioStreamId: Int? = null
+    private var rtmpVideoStreamId: Int? = null
+    private var fileAudioStreamId: Int? = null
+    private var fileVideoStreamId: Int? = null
+
+    // Keep video configuration separate for rtmp and file
+    private var rtmpVideoConfig: VideoConfig? = null
+    private var fileVideoConfig: VideoConfig? = null
+    // share the same audio config (and the same encoder)
+    private var audioConfig: AudioConfig? = null
+
+    private val sourceOrientationProvider: ISourceOrientationProvider
+        get() = cameraSource.orientationProvider
+
+    override val onInternalErrorListener = object : OnErrorListener {
+        override fun onError(error: StreamPackError) {
+            onStreamError(error)
+        }
+    }
+
+    private val audioEncoderListener = object : IEncoderListener {
+        override fun onInputFrame(buffer: ByteBuffer): Frame {
+            return audioSource.getFrame(buffer)
+        }
+
+        override fun onOutputFrame(frame: Frame) {
+            val fileFrame = frame.clone()
+            rtmpAudioStreamId?.let {
+                try {
+                    this@CameraDualLiveStreamer.rtmpMuxer.encode(frame, it)
+                } catch (e: Exception) {
+                    throw StreamPackError(e)
+                }
+            }
+            fileAudioStreamId?.let {
+                try{
+                    this@CameraDualLiveStreamer.fileMuxer.encode(fileFrame,it)
+                } catch(e:Exception){
+                    throw StreamPackError(e)
+                }
+            }
+        }
+    }
```

Comment:
per offline discussion, this construction implies that the audio encoding (and thus audio settings) are shared between the different output destinations. This is something that I can live with for my needs, since it allows for a much simpler implementation.